js/services/Vocabulary.js

```diff
@@ -12,39 +12,47 @@ define(function (require, exports) {
 	loadedPromise.resolve();
 
 	var defaultSource;
-	var domainPromise = $.Deferred();
 	var domains = [];
```

Comment:
Why global variable is still used?

Reply:
Yes, we will want a global variable to hold the promise.  I am pushing up a commit to demonstrate this.